contracts/protocol/tokenization/StaticATokenLM.sol

```diff
@@ -70,6 +70,9 @@ contract StaticATokenLM is
 
   bool public isImplementation;
 
+  address private messagingContract;
+  address private l2TokenBridge;
```

Comment:
Why store two pieces of state instead of implementing the message send in the `l1TokenBridge`?

Reply:
I started doing that, then I wondered if it is better to interact with L1 token bridge or if it is better to be agnostic to it. I chosed the second option, but here we're not agnostic with L2 token bridge. I'll move the logic to L1 token bridge and adapt the code here.